### PR TITLE
BITMAKER-2638 Change a user email (frontend & backend)

### PR DESCRIPTION
**Pull request by webtaken**
# Description

Add feature to change email associated with an user.
To perform this action is necessary password.

Acceptance criteria:
Users can change email associated with their accounts.
Note: This feature is implemented only in frontend, ticket for backend is this https://tasks.bitmaker.dev/issues/2127.

# Issue

* https://tasks.bitmaker.dev/issues/2638.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines of this project.
- [x] I have made corresponding changes to the [documentation](https://github.com/bitmakerla/estela/tree/main/docs).
- [x] New and existing tests pass locally with my changes.
- [x] If this change is a core feature, I have added thorough tests.
- [x] If this change affects or depends on the behavior of other _estela_ repositories, I have created pull requests with the relevant changes in the affected repositories. Please, refer to our [official documentation](https://estela.bitmaker.la/docs/).
- [x] I understand that my pull request may be closed if it becomes obvious or I did not perform all of the steps above.

### CODE REVIEW

File: estela-web/src/shared/notifications/index.tsx [hunk 1]
```diff
@@ -10,6 +10,13 @@ export const authNotification = (): void => {
     });
 };
 
+export const apiCallFail = (): void => {
```
Comment (reqhiem) [inline estela-web/src/shared/notifications/index.tsx:13]: @webtaken Is this resource being used?

File: estela-api/api/views/auth.py
Comment (reqhiem) [inline estela-api/api/views/auth.py:1]: Line 161, comparison to None should be 'if cond is None'

File: estela-api/api/serializers/auth.py
Comment (reqhiem) [inline estela-api/api/serializers/auth.py:1]: Line 2, imported but unused

**Review by reqhiem — CHANGES_REQUESTED** [review]

File: estela-web/src/shared/notifications/index.tsx (hunk 1 above)
Comment (webtaken) [reply estela-web/src/shared/notifications/index.tsx]: I deleted the notification, I didn't see it in any document.

File: estela-api/api/views/auth.py
Comment (webtaken) [reply estela-api/api/views/auth.py:1]: Yes that's better, I changed it, thanks for the correction.

File: estela-api/api/serializers/auth.py
Comment (webtaken) [reply estela-api/api/serializers/auth.py:1]: Removed unused import.

**Review by reqhiem — APPROVED** [review]
GJ 👍🏻

**Comment by reqhiem** [comment]
@emegona This PR will be merged. Could you take a look at it?

File: development-helps/Makefile [hunk 2]
```diff
@@ -114,6 +113,9 @@ docs:
 	-cd $(ROOT_DIR) && \
 	 export CREDENTIALS=local && export ENGINE=kubernetes && export SPIDERDATA_DB_ENGINE=mongodb && \
 	 python3 estela-api/manage.py generate_swagger -f yaml estela-api/$(API_DOC)
+	-rm -f $(API_DOC_WEB)
```
Comment (emegona) [inline development-helps/Makefile:116]: Where is `API_DOC_WEB` defined?

File: estela-api/api/serializers/auth.py [hunk 3]
```diff
@@ -76,3 +75,30 @@ class UserProfileSerializer(serializers.HyperlinkedModelSerializer):
...
+            UniqueValidator(
+                queryset=User.objects.all(),
+                message="A user with that email already exists",
+            )
+        ]
+    )
+
+    class Meta:
+        model = User
+        fields = ["username", "email", "password", "last_password_change"]
```
Comment (emegona) [inline estela-api/api/serializers/auth.py:104]: We are duplicating the code here. `UserProfileUpdateSerializer` is very similar to `UserProfileSerializer`. If you want not to show the `password` field, you can set it to `write-only` so it is not returned in the API response.

**Review by emegona — CHANGES_REQUESTED** [review]
@webtaken Please address my comments. Thank you.

File: development-helps/Makefile (hunk 2 above)
Comment (webtaken) [reply development-helps/Makefile]: I deleted that line, now works fine.

File: estela-api/api/serializers/auth.py (hunk 3 above)
Comment (webtaken) [reply estela-api/api/serializers/auth.py:80]: Thanks, deleted UserProfileUpdateSerializer and added password field with `write_only=True` to the UserProfileSerializer

**Review by emegona — APPROVED** [review]

**Pull request merged**